### PR TITLE
Upgrade tests: reexamine cross-testing matrix

### PR DESCRIPTION
**Pull request by edsantiago**
- removed: v1.9.0, v2.0.6
 + added:   v3.4.0

(Cannot add v4 because there's no such image on quay. As soon
as one appears, we should add it.)

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -721,14 +721,12 @@ upgrade_test_task:
...
-        - env:
-              PODMAN_UPGRADE_FROM: v1.9.0
-        - env:
-              PODMAN_UPGRADE_FROM: v2.0.6
         - env:
               PODMAN_UPGRADE_FROM: v2.1.1
         - env:
               PODMAN_UPGRADE_FROM: v3.1.2
+        - env:
+              PODMAN_UPGRADE_FROM: v3.4.0
```
Comment (edsantiago) [inline .cirrus.yml:729]: quay also has v3.4.1, 2, and 4. Should we use one of those instead?

Comment (Luap99) [reply .cirrus.yml:729]: I would go with .4 to have more bug fixes in it.

File: .cirrus.yml [hunk 2]
```diff
@@ -721,14 +721,12 @@ upgrade_test_task:
     depends_on:
       - local_system_test
     matrix:
-        - env:
-              PODMAN_UPGRADE_FROM: v1.9.0
```
[the remark this answers is not in the document]
Comment (edsantiago) [reply .cirrus.yml:725]: We discussed this a week or two ago, and my recollection was that nobody would ever upgrade from v1 to v4. But I may be wrong. Does anyone remember that discussion?

[the remark this answers is not in the document]
Comment (mheon) [reply .cirrus.yml:725]: I think we want to support stable version -> stable version - so 1.6.4 to 3.0 would be my expectation of the oldest upgrade we'd reasonably expect.

**Comment by edsantiago** [comment]
`podman pod create [no longer works in v3.4.x](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4852462518534144/html/upgrade_test-podman-fedora-35-root-host.log.html) (containerized). Simple reproducer:
```console
$ sudo bin/podman run --rm --privileged --net=host quay.io/podman/stable:v3.4.0 podman pod create --name foo
Error: invalid config provided: cannot set hostname when running in the host UTS namespace: invalid configuration
```
(works with `v3.1.2`).

Any suggestions on how to fix that?

**Comment by edsantiago** [comment]
Issue resolved, thanks to Dan https://github.com/containers/podman/issues/11969#issuecomment-943386484

**Comment by Luap99** [comment]
> Issue resolved, thanks to Dan [#11969 (comment)](https://github.com/containers/podman/issues/11969#issuecomment-943386484)

This sounds like an ugly workaround. If our image does not support `podman pod create` out of the box it is broken!

**Comment by edsantiago** [comment]
I was thinking the same thing, but ass-u-med there must be a good reason for the default.

**Comment by Luap99** [comment]
I cannot think of any reason why the uts namespace is set to host by default. I see reasons using host for cgroup and netns but not the other namespaces.

**Comment by edsantiago** [comment]
As best I can tell, it dates to [PR 5467](https://github.com/containers/podman/pull/5647/files#diff-236d43322cdc268992781da4a6d2c1ed11157fad3fda1f024481cb108e2244f6) which added the `containers.conf` file. I can't tell if the presence of `utsns` is intentional or an oversight; and this is not something we can resolve in this PR. Would you like to file an issue (or PR)? Ordinarily I would do so but I don't understand the problem well enough to describe it properly.

**Comment by Luap99** [comment]
Created https://github.com/containers/podman/issues/13714, lets move the discussion there.

**Comment by edsantiago** [comment]
@containers/podman-maintainers this is as ready as I can make it.

**Pull request merged**